src/script/script_config.cpp

```diff
@@ -31,15 +31,15 @@ void ScriptConfig::Change(const char *name, int version, bool force_exact_match,
 
 	this->ClearConfigList();
 
-	if (_game_mode == GM_NORMAL && this->info != nullptr) {
+	if (_game_mode == GM_NORMAL && _switch_mode != SM_LOAD_GAME && this->info != nullptr) {
 		/* If we're in an existing game and the Script is changed, set all settings
 		 *  for the Script that have the random flag to a random value. */
 		for (ScriptConfigItemList::const_iterator it = this->info->GetConfigList()->begin(); it != this->info->GetConfigList()->end(); it++) {
 			if ((*it).flags & SCRIPTCONFIG_RANDOM) {
 				this->SetSetting((*it).name, InteractiveRandomRange((*it).max_value + 1 - (*it).min_value) + (*it).min_value);
 			}
 		}
-		this->AddRandomDeviation();
+		this->AddRandomDeviation(false);
```

Comment:
Why is the entire changeset even this large? Wouldn't it make more sense to just not call AddRandomDeviation at all if the script that's being restarted after load has already been started? So scripts that were active when the save was made have their settings static (all of them), and scripts that were not started yet can get randomized.

Reply:
I'm not sure how to answer this.

The scripts are not started at the time ScriptConfig::Change is called from ai_sl.cpp / game_sl.cpp.

Fix from commit 1 already prevents AddRandomDeviation from being called after load. So I'm confused as to what you're requesting here.